tools/autograd/gen_python_functions.py

```diff
@@ -245,190 +239,157 @@ def group_declarations_by_op_name(declarations):
+            #
+            # python_nn_functions.cpp.obj : error LNK2001: unresolved external symbol
+            # PyStructSequence_UnnamedField
+            #
+            # Thus, at this point in time, we do not support unnamed
+            # fields in namedtuple; you must either name all fields,
+            # or none of them.
+            raise ValueError("Unnamed field is not supported by codegen")
+
+        return list(map(lambda r: str(r.name), returns))
+
+@with_native_function
+def gen_namedtuple_typename_key(f: NativeFunction) -> str:
+    name = cpp.name(f.func)
+    fieldnames = namedtuple_fieldnames(f)
+    return '_'.join([name] + fieldnames)
+
+def emit_namedtuple_typedefs(
+    overloads: Tuple[PythonSignatureNativeFunctionPair, ...]
+) -> Tuple[List[str], Dict[str, str]]:
     """
     Generate block of named tuple type def inits, and add typeref snippets
     to declarations that use them
     """
-    flddefnames = {}    # map from unique field name lists to field def name
-    flddefs = []        # field def declarations
-    typenames = {}      # map from unique name + field name lists to typedef name
-    typedefs = []       # typedef declarations and init code
-
-    for decl in declarations:
-        fieldnames = namedtuple_fieldnames(decl)
-        if fieldnames == []:
-            decl['namedtuple_typeref'] = ''
+    flddefnames: Dict[str, str] = {}  # map from unique field name lists to field def name
+    flddefs: List[str] = []           # field def declarations
+    typenames: Dict[str, str] = {}    # map from unique name + field name lists to typedef name
+    typedefs: List[str] = []          # typedef declarations and init code
+
+    for overload in overloads:
+        fieldnames = namedtuple_fieldnames(overload.function)
+        if not fieldnames:
             continue
 
         fn_key = '_'.join(fieldnames)
         fieldsname = flddefnames.get(fn_key)
         if fieldsname is None:
-            fieldsname = 'NamedTuple_fields{}'.format('' if flddefs == [] else len(flddefs))
-            fields = ['{{"{}", ""}}'.format(fn) for fn in fieldnames]
-            fieldsdef = PY_NAMEDTUPLE_FIELDSDEF.substitute(
-                fieldsname=fieldsname,
-                fields=fields
-            )
+            fieldsname = f'NamedTuple_fields{"" if not flddefs else len(flddefs)}'
             flddefnames[fn_key] = fieldsname
-            flddefs.append(fieldsdef)
+            fields = ', '.join(f'{{"{fn}", ""}}' for fn in fieldnames)
+            flddefs.append(f"""\
+static PyStructSequence_Field {fieldsname}[] = {{ {fields},  {{nullptr}} }};
+""")
 
-        name = decl['name']
-        key = '{}_{}'.format(name, '_'.join(fieldnames))
-        typename = typenames.get(key)
+        name = cpp.name(overload.function.func)  # use @with_native_function?
+        tn_key = gen_namedtuple_typename_key(overload.function)
+        typename = typenames.get(tn_key)
```

Comment:
It would make me feel better if there was some asserts here that would tell me that order of overloads doesn't matter for these initializations but I can't think of an easy way to actually do this.